packages/ui/src/bounty/modals/ContributeFundsModal/ContributeFundsModal.tsx

```diff
@@ -48,45 +55,47 @@ export const ContributeFundsModal = () => {
     showModal,
   } = useModal<BountyContributeFundsModalCall>()
   const { api, isConnected } = useApi()
-  const minFundingLimit = api?.consts.bounty.minFundingLimit.toNumber() ?? 0
+  const [state, send] = useMachine(contributeFundsMachine)
+  const minFundingLimit = api?.consts.bounty.minFundingLimit ?? BN_ZERO
   const { active: activeMember } = useMyMemberships()
   const { allAccounts } = useMyAccounts()
-  const [amount, setAmount] = useState<string>(String(minFundingLimit))
-  const [state, send] = useMachine(contributeFundsMachine)
-  const [account, setAccount] = useState<Account>()
-  const balance = useBalance(account?.address)
+  const balance = useBalance(activeMember?.controllerAccount)
 
-  const setStakingAmount = useCallback((_, value: number) => setAmount(String(value)), [])
+  const setAmount = useCallback((amount: BN | string) => send('SET_AMOUNT', { amount: new BN(amount) }), [])
 
-  const valid = useMemo(() => new BN(amount).gten(minFundingLimit) && !!account, [amount, account])
+  const { setContext, isValid, errors } = useSchema({ amount: state.context.amount }, schema)
+  useEffect(() => {
+    if (minFundingLimit) {
+      setContext({
+        minAmount: minFundingLimit,
+      })
+
+      if (!state.context.amount && state.matches(ContributeFundStates.contribute)) {
+        setAmount(minFundingLimit.toString())
+      }
```

Comment:
Is this one needed since `minFundingLimit` is already used as a fallback in the transaction ?

Reply:
This one is just about pre populating input with minimal allowed value, there isn't really connection with tx fallback.